src/triangle.rs

```diff
@@ -0,0 +1,344 @@
+    pub fn inflate(&self, scalar: f64) -> Self {
+        let centroid = self.centroid();
+
+        Self::new(
+            centroid + (0.0, scalar),
+            centroid + scalar * Vec2::from_angle(5.0 * FRAC_PI_4),
+            centroid + scalar * Vec2::from_angle(7.0 * FRAC_PI_4),
+        )
+    }
+
+    /// Is this [`Triangle`] finite?
+    #[inline]
+    pub fn is_finite(&self) -> bool {
+        self.a.is_finite() && self.b.is_finite() && self.c.is_finite()
+    }
+
+    /// Is this [`Triangle`] NaN?
+    #[inline]
+    pub fn is_nan(&self) -> bool {
+        self.a.is_nan() || self.b.is_nan() || self.c.is_nan()
+    }
+
+    /// [`Triangle`]'s vertices as an array of Points
+    #[inline]
+    pub fn as_array(&self) -> [Point; 3] {
+        [self.a, self.b, self.c]
+    }
+}
+
+impl From<(Point, Point, Point)> for Triangle {
+    fn from(points: (Point, Point, Point)) -> Triangle {
+        Triangle::new(points.0, points.1, points.2)
+    }
+}
+
+impl From<(Line, Line)> for Triangle {
+    fn from(lines: (Line, Line)) -> Triangle {
+        Triangle::new(lines.0.p0, lines.0.p1, lines.1.p0)
+    }
+}
+
+impl Add<Vec2> for Triangle {
+    type Output = Triangle;
+
+    #[inline]
+    fn add(self, v: Vec2) -> Triangle {
+        Triangle::new(self.a + v, self.b + v, self.c + v)
+    }
+}
+
+impl Sub<Vec2> for Triangle {
+    type Output = Triangle;
+
+    #[inline]
+    fn sub(self, v: Vec2) -> Triangle {
+        Triangle::new(self.a - v, self.b - v, self.c - v)
+    }
+}
+
+#[doc(hidden)]
+pub struct TrianglePathIter {
+    triangle: Triangle,
+    ix: usize,
+}
+
+impl Shape for Triangle {
+    type PathElementsIter<'iter> = TrianglePathIter;
+
+    fn path_elements(&self, _tolerance: f64) -> TrianglePathIter {
+        TrianglePathIter {
+            triangle: *self,
+            ix: 0,
+        }
+    }
+
+    #[inline]
+    fn area(&self) -> f64 {
+        Triangle::area(self)
+    }
+
+    #[inline]
+    fn perimeter(&self, _accuracy: f64) -> f64 {
+        self.a.distance(self.b) + self.b.distance(self.c) + self.c.distance(self.a)
+    }
+
+    #[inline]
+    fn winding(&self, pt: Point) -> i32 {
+        let s0 = (self.b - self.a).cross(pt - self.a).signum();
+        let s1 = (self.c - self.b).cross(pt - self.b).signum();
+        let s2 = (self.a - self.c).cross(pt - self.c).signum();
+
+        if s0 == s1 && s1 == s2 {
+            s0 as i32
+        } else {
+            0
+        }
+    }
+
+    #[inline]
+    fn bounding_box(&self) -> Rect {
+        Rect::new(
+            self.a.x.min(self.b.x.min(self.c.x)),
+            self.a.y.min(self.b.y.min(self.c.y)),
+            self.a.x.max(self.b.x.max(self.c.x)),
+            self.a.y.max(self.b.y.max(self.c.y)),
+        )
+    }
+
+    #[inline]
+    fn as_rect(&self) -> Option<Rect> {
+        Some(self.bounding_box())
+    }
+}
+
+// NOTE: vertices a, b and c are not garunteed to be in order as described in the struct comments
```

Comment:
"guaranteed"

Reply:
thanks, fixed in [8da1a8c](https://github.com/linebender/kurbo/pull/350/commits/8da1a8c506fa99d3ca76ed36b39986d157143734)!

a bit unrelated but do you think this justifies my `organise` proposal ([8b03193](https://github.com/linebender/kurbo/pull/350/commits/8b03193f6f932807b723c9d693a4010ba5e5076d))?

Reply:
Has this been submitted to [typos](https://github.com/crate-ci/typos)?

Reply:
yep! https://github.com/crate-ci/typos/issues/1024#issuecomment-2168557914